src/pyfmi/fmi3.pyx

```diff
@@ -689,6 +689,36 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
         variable_data_type = self._get_variable_data_type(variable_name)
         return FMI3_Type(int(variable_data_type))
 
+    cpdef get_variable_description(self, variable_name):
+        """
+        Get the description of a given variable.
+
+        Parameter::
+
+            variable_name --
+                The name of the variable
+
+        Returns::
+
+            The description of the variable.
+        """
+        cdef FMIL3.fmi3_import_variable_t* variable
+        cdef FMIL3.fmi3_string_t desc
+
+        variable_name = pyfmi_util.encode(variable_name)
+        cdef char* var_name = variable_name
+
+        variable = FMIL3.fmi3_import_get_variable_by_name(self._fmu, var_name)
+        if variable == NULL:
+            raise FMUException("The variable %s could not be found." % var_name)
```

Comment:
Does this format properly with all the encode/decode?

Might be worth adding a unit test for these?

(Likely affects a few other places as well)

Reply:
Changed in b384efd, see also added test (and updated tests)